api/migrations/20200608243419-removeFeeOrders.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+let dbm;
+let type;
+let seed;
+
+/**
+  * We receive the dbmigrate dependency from dbmigrate initially.
+  * This enables us to not have to rely on NODE_PATH.
+  */
+exports.setup = function(options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;;
+  seed = seedLink
+};
+
+exports.up = async function(db) {
+  const mClient = await db.connection.connect(db.connectionString, { native_parser: true });
+
+  try {
+    const nrptiCollection = await mClient.collection('nrpti');
+
+    const orders = await nrptiCollection.find({ _schemaName: 'Order' }).toArray();
+    const inspections = await nrptiCollection.find({ _schemaName: 'Inspection' }).toArray();
+
+    const recordToCheck = [...orders, ...inspections];
+
+    // Filter out only the fee orders  
+    // Any document names that contain these terms are considered Fee Orders.
+    const orderTermsBlacklist = [
+      'fee order',
+      'order to pay fees',
+      'fee package'
+    ];
+
+    const feeOrders = [];
+
+    for (const record of recordToCheck) {
+      for (const term of orderTermsBlacklist) {
+        if (record.recordName.toLowerCase().includes(term)) {
+          feeOrders.push(record);
+          break;
+        }
+      }
+    }
+
+    console.log(`${feeOrders.length} Fee Orders found.`);
+
+    for (const order of feeOrders) {
+      // Delete any flavour records.
+      for (const flavourId of order._flavourRecords) {
+        nrptiCollection.remove({ _id: flavourId });
+      }
+
+      // Delete any document records.
+      for (const documentId of order.documents) {
+        nrptiCollection.remove({ _id: documentId });
```

Comment:
We should also check for S3 documents

pseudo code:
```
const document = await Document.findOneAndRemove(...)
if(document.key)
then  DocumentController.deleteS3Document(key)
```

Reply:
I checked an it looks to me that all Epic documents are linked back to Epic's own URLs and we are not hosting any. Is that correct or do we indeed have some?

Reply:
All of the records imported from Epic will only be link documents.
The only way you'd get s3 documents is if an admin manually added one.

So there probably just aren't any

Reply:
Okay, will update to include that check just in case.